tests-arquillian/src/test/java/org/jboss/weld/tests/jsf/flow/FlowMultipleWarTest.java

```diff
@@ -0,0 +1,117 @@
+
+/**
+ * See also WELD-2448. If multiple deployments share a proxy class from static module (JSF impl in this test case) and each deployment uses a different value
+ * for {@link ConfigurationKey#RELAXED_CONSTRUCTION} the proxy instance might not be initialized corretly.
+ *
+ * @author Martin Kouba
+ */
+@Category(Integration.class)
+@RunWith(Arquillian.class)
+public class FlowMultipleWarTest {
+
+    private static final String WAR1 = "war1";
+
+    private static final String WAR2 = "war2";
+
+    @Deployment(name = WAR1, managed = false, testable = false)
+    public static WebArchive createFooTestArchive() {
+        WebArchive war1 = ShrinkWrap.create(WebArchive.class, WAR1 + ".war")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "index.xhtml",
+                        "index.xhtml")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "simple-flow-flow.xml",
+                        "simple-flow/simple-flow-flow.xml")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "simple-flow.xhtml",
+                        "simple-flow/simple-flow.xhtml")
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml").addClass(Foo.class)
+                .setWebXML(FlowMultipleWarTest.class.getPackage(), "web.xml");
+        return war1;
+    }
+
+    @Deployment(name = WAR2, managed = false, testable = false)
+    public static WebArchive createBarTestArchive() {
+        return ShrinkWrap.create(WebArchive.class, WAR2 + ".war")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "index.xhtml",
+                        "index.xhtml")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "simple-flow-flow.xml",
+                        "simple-flow/simple-flow-flow.xml")
+                .addAsWebResource(FlowMultipleWarTest.class.getPackage(), "simple-flow.xhtml",
+                        "simple-flow/simple-flow.xhtml")
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml").addClass(Bar.class)
+                .setWebXML(FlowMultipleWarTest.class.getPackage(), "web.xml").addAsResource(
+                        PropertiesBuilder.newBuilder()
+                                .set(ConfigurationKey.RELAXED_CONSTRUCTION.get(), "true").build(),
+                        "weld.properties");
+    }
+
+    @ArquillianResource
+    Deployer deployer;
+
+    /**
+     * This is not a real test method - we just need to deploy WAR1 first.
+     */
+    @Test
+    @RunAsClient
+    @InSequence(1)
+    public void deployArchives() {
+        // In order to use @ArquillianResource URLs we need to deploy both test archives first
+        deployer.deploy(WAR1);
+        deployer.deploy(WAR2);
+    }
+
+    @Test
+    @RunAsClient
+    @InSequence(2)
+    public void testFlows(@ArquillianResource @OperateOnDeployment(WAR1) URL war1Context,
+            @ArquillianResource @OperateOnDeployment(WAR2) URL war2Context)
```

Comment:
Isn't this test missing the part where you invoke the same call from war2Context? Or how do you make sure the extension proxy get initiated from the other WAR as well?

Reply:
I think it should be initialized during bootstrap of a JSF app with flows.

Reply:
Hmm, makes sense.
If that's the case then the second method param is useless and can be removed?